client/blocks/credit-card/payment-method.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import { Elements, useStripe } from '@stripe/react-stripe-js';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getStripeServerData } from '../stripe-utils';
+import { useCheckoutSubscriptions } from './use-checkout-subscriptions';
+import { InlineCard, CardElements } from './elements';
+
+/**
+ * @typedef {import('../stripe-utils/type-defs').Stripe} Stripe
+ * @typedef {import('../stripe-utils/type-defs').StripePaymentRequest} StripePaymentRequest
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ */
+
+export const getStripeCreditCardIcons = () => {
+	return Object.entries( getStripeServerData().icons ).map(
+		( [ id, { src, alt } ] ) => {
+			return {
+				id,
+				src,
+				alt,
+			};
+		}
+	);
+};
+
+/**
+ * Stripe Credit Card component
+ *
+ * @param {RegisteredPaymentMethodProps} props Incoming props
+ */
+const CreditCardComponent = ( {
+	billing,
+	eventRegistration,
+	emitResponse,
+	components,
+} ) => {
+	const { ValidationInputError, PaymentMethodIcons } = components;
+	const [ sourceId, setSourceId ] = useState( '' );
+	const stripe = useStripe();
+	const onStripeError = useCheckoutSubscriptions(
+		eventRegistration,
+		billing,
+		sourceId,
+		setSourceId,
+		emitResponse,
+		stripe
+	);
+	const onChange = ( paymentEvent ) => {
+		if ( paymentEvent.error ) {
+			onStripeError( paymentEvent );
+		}
+		setSourceId( '0' );
```

Comment:
@nerrad Is there a reason we set the source ID to `'0'` here, as opposed to `''` like it's initialized?

I'm just wondering if there's a good reason for this, because this could mess with code where we check the "truthiness" of `sourceId`. While `'0'` is not an empty string, JS interprets it as falsey (`'0' == false`), and that could mess with some of our assumptions about the contents of `sourceId`.

Reply:
I honestly can't remember. Your rationale makes sense and if you want to do that change and update the affected logic checks everywhere, that makes sense to me.

Reply:
I'll look into it, thank you!

Reply:
We now set sourceId to `''` as of cc7ba9486f397a0c87b9c154c02717850aee7649, since setting it to empty string doesn't break anything, and makes more sense IMO since the initial value is also an empty string.